youtube_dl/YoutubeDL.py

```diff
@@ -1237,6 +1237,10 @@ def process_video_result(self, info_dict, download=True):
         if thumbnails and 'thumbnail' not in info_dict:
             info_dict['thumbnail'] = thumbnails[-1]['url']
 
+        if self.params.get('list_thumbnails'):
+            self.list_thumbnails(info_dict)
+            return
```

Comment:
This can even go before `if` block at 1237 if we remove references to `thumbnail` (1905-1908) in `list_thumbnails`. Since `thumbnails` will always be present at that point when at least one thumbnail is extracted.